src/mon/OSDMonitor.cc

```diff
@@ -14046,6 +14046,16 @@ bool OSDMonitor::prepare_pool_op(MonOpRequestRef op)
 
   const pg_pool_t *pool = osdmap.get_pg_pool(m->pool);
 
+  if (m->op == POOL_OP_CREATE_SNAP ||
+      m->op == POOL_OP_CREATE_UNMANAGED_SNAP) {
+    if (const auto& fsmap = mon.mdsmon()->get_fsmap(); fsmap.pool_in_use(m->pool)) {
+      dout(20) << "pool level snapshots have been disabled for pools "
```

Comment:
How about "monitor-managed" instead of "pool level". That keeps making me think of "pool snapshots", which these aren't.

Reply:
hmm, okay
btw, `ceph osd pool mksnap <pool> <snap>` and `rados --pool=<pool> mksnap <snap>` both land here
that made me think they are the same

Also, I'm probably lacking a ton of context and insight into the workings.

Reply:
Those *are* the same. But what also lands here is when rbd sends a mksnap for the "self-managed" interface, which is by far the most common user. :)

"Pool snapshots" themselves are almost dead; none of our applications use them and we recommend against them -- we have better interfaces and pool snapshots are generated by osdmap propagation rather than being point-in-time like rbd or cephfs snaps are.

Reply:
replaced "pool level snapshots" with "monitor manged snapshots"